### PR TITLE
Add codecov

### PR DESCRIPTION
**Pull request by lucianopaz**
This PR adds a step to CI that uploads the xml coverage report to codecov

### CODE REVIEW

**Comment by ricardoV94** [comment]
Github complains about something: 

image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,5 +37,11 @@ jobs:
...
         run: |
-          make init
-          make test
+          pip install -e .[test]
+          pytest --cov-report=xml --no-cov-on-fail
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # use token for more robust uploads
+          name: ${{ python-version }}
```
[the remark this answers is not in the document]
Comment (lucianopaz) [reply .github/workflows/ci.yml:46]: Thanks Ricardo!

**Comment by ricardoV94** [comment]
Seems to not be uploading it? https://github.com/pymc-labs/pymc-marketing/actions/runs/4184385579/jobs/7249964095

**Comment by lucianopaz** [comment]
@twiecki, it looks like codecov hasn't been green lighted for the pymc-labs org yet. Could you have a look?

image

**Comment by twiecki** [comment]
Weird, I granted perms. Maybe we merge and see if we're lucky?

**Comment by lucianopaz** [comment]
> Weird, I granted perms. Maybe we merge and see if we're lucky?

We can merge now. I saw that the org started to show up on codecov and generate a new token for the repo there.

**Comment by lucianopaz** [comment]
I suppose that once this pr gets merged, the token will let the codecov app kick in so that if comments the coverage of PRs

**Pull request merged**